### PR TITLE
Report cannot move errors in promoted MIR

### LINKED ISSUE #70934: ICE: No `move_errors` should be allowed in MIR borrowck
I get an ICE when I try to build [evolvim](https://github.com/syberant/evolvim):
```
error: internal compiler error: No `move_errors` should be allowed in MIR borrowck
   --> evolvim-lib\./src/lib\neat\genome\mod.rs:100:13
    |
100 |             InputType::Eye(EYE[2]).clone(),
    |             ^^^^^^^^^^^^^^^^^^^^^^

error: internal compiler error: No `move_errors` should be allowed in MIR borrowck
  --> evolvim-lib\./src/lib\neat\genome\mod.rs:99:13
   |
99 |             InputType::Eye(EYE[1]).clone(),
   |             ^^^^^^^^^^^^^^^^^^^^^^

error: internal compiler error: No `move_errors` should be allowed in MIR borrowck
  --> evolvim-lib\./src/lib\neat\genome\mod.rs:98:13
   |
98 |             InputType::Eye(EYE[0]).clone(),
   |             ^^^^^^^^^^^^^^^^^^^^^^

thread 'rustc' panicked at 'no errors encountered even though `delay_span_bug` issued', src\librustc_errors\lib.rs:357:17
stack backtrace:
   0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
   1: core::fmt::write
   2: <std::io::IoSliceMut as core::fmt::Debug>::fmt
   3: std::panicking::take_hook
   4: std::panicking::take_hook
   5: rustc_driver::report_ice
   6: std::panicking::rust_panic_with_hook
   7: <rustc_errors::snippet::Style as core::fmt::Debug>::fmt
   8: <rustc_errors::HandlerInner as core::ops::drop::Drop>::drop
   9: rustc_driver::pretty::print_after_hir_lowering
  10: rustc_driver::pretty::print_after_hir_lowering
  11: rustc_driver::pretty::print_after_hir_lowering
  12: rustc_driver::pretty::print_after_hir_lowering
  13: <env_logger::filter::inner::Filter as core::fmt::Display>::fmt
  14: <env_logger::filter::inner::Filter as core::fmt::Display>::fmt
  15: <env_logger::filter::inner::Filter as core::fmt::Display>::fmt
  16: _rust_maybe_catch_panic
  17: rustc_driver::pretty::print_after_hir_lowering
  18: ZN244_$LT$std..error..$LT$impl$u20$core..convert..From$LT$alloc..string..String$GT$$u20$for$u20$alloc..boxed..Box$LT$dyn$u20$std..error..Error$u2b$core..marker..Sync$u2b$core..marker..Send$GT$$GT$..from..StringError$u20$as$u20$core..fmt..Display$GT$3fmt17
  19: std::sys::windows::thread::Thread::new
  20: BaseThreadInitThunk
  21: RtlUserThreadStart
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.
```
I tried stable, beta and nightly. All ended with an ICE.
```
>rustc --version --verbose
rustc 1.42.0 (b8cedc004 2020-03-09)
binary: rustc
commit-hash: b8cedc00407a4c56a3bda1ed605c6fc166655447
commit-date: 2020-03-09
host: x86_64-pc-windows-msvc
release: 1.42.0
LLVM version: 9.0
```

### PR DESCRIPTION
**Pull request by matthewjasper**
Closes #70934

### CODE REVIEW

File: src/librustc_mir/borrow_check/mod.rs [hunk 1]
```diff
@@ -264,6 +267,41 @@ fn do_mir_borrowck<'a, 'tcx>(
...
+                borrow_set: borrow_set.clone(),
+                dominators,
+                upvars: Vec::new(),
+                local_names: IndexVec::from_elem(None, &promoted_body.local_decls),
+                region_names: RefCell::default(),
+                next_region_name: RefCell::new(1),
+                polonius_output: None,
+            };
+            promoted_mbcx.report_move_errors(move_errors);
+            errors_buffer = mem::take(&mut promoted_mbcx.errors_buffer)
```
Comment (nikomatsakis) [inline src/librustc_mir/borrow_check/mod.rs:301]: Do we really want to *overwrite* `errors_buffer` each time, and not (e.g.) something like

```rust
errors_buffer.extend(promoted_mbcx.errors_buffer.drain(..));
```

Comment (matthewjasper) [reply src/librustc_mir/borrow_check/mod.rs:301]: Yes, `errors_buffer` is being moved into `promoted_mbcx`

**Pull request merged**